workspaces/src/result.rs

```diff
@@ -334,7 +334,7 @@ impl ExecutionSuccess {
 
 impl<T> ExecutionResult<T> {
     /// Returns just the transaction outcome.
-    pub fn outcome(&self) -> &ExecutionOutcome {
+    pub const fn outcome(&self) -> &ExecutionOutcome {
```

Comment:
Since you cannot get outcome without making a request, I believe it is quite pointless to make this method `const`. Removing `const` in the future will be breaking change, so I would avoid unnecessary `const` markers.

P.S. Apply the same reasoning to all the other methods as I feel that half of the `const` methods should not be `const` really.